File: docs/snippets/all/archetypes/graph_directed.cpp
//! Log a simple directed graph.

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_graph_directed");
    rec.spawn().exit_on_failure();

    rec.log(
        "simple",
        rerun::GraphNodes({"a", "b", "c"})
            .with_positions({{0.0, 100.0}, {-100.0, 0.0}, {100.0, 0.0}})
            .with_labels({"A", "B", "C"}),
        rerun::GraphEdges({{"a", "b"}, {"b", "c"}, {"c", "a"}})
            // Graphs are undirected by default.
            .with_graph_type(rerun::components::GraphType::Directed)
    );
}

File: docs/snippets/all/archetypes/graph_directed.py
"""Log a simple directed graph."""

import rerun as rr

rr.init("rerun_example_graph_directed", spawn=True)

rr.log(
    "simple",
    rr.GraphNodes(
        node_ids=["a", "b", "c"], positions=[(0.0, 100.0), (-100.0, 0.0), (100.0, 0.0)], labels=["A", "B", "C"]
    ),
    rr.GraphEdges(edges=[("a", "b"), ("b", "c"), ("c", "a")], graph_type="directed"),
)
